File: app/src/main/java/com/example/musicwiki/repo/network/albumsinfo/Track.java

package com.example.musicwiki.repo.network.albumsinfo;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Track {

    @SerializedName("streamable")
    @Expose
    private Streamable streamable;
    @SerializedName("duration")
    @Expose
    private Integer duration;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("@attr")
    @Expose
    private Attr attr;
    @SerializedName("artist")
    @Expose
    private Artist artist;

    public Streamable getStreamable() {
        return streamable;
    }

    public void setStreamable(Streamable streamable) {
        this.streamable = streamable;
    }

    public Integer getDuration() {
        return duration;
    }

    public void setDuration(Integer duration) {
        this.duration = duration;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Attr getAttr() {
        return attr;
    }

    public void setAttr(Attr attr) {
        this.attr = attr;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

}

File: app/src/main/java/com/example/musicwiki/repo/network/toptracks/Track.kt
package com.example.musicwiki.repo.network.toptracks

import com.google.gson.annotations.SerializedName
import com.google.gson.annotations.Expose
import com.example.musicwiki.repo.network.toptracks.Streamable

data class Track(
    @SerializedName("name")
    @Expose
    var name: String? = null,

    @SerializedName("duration")
    @Expose
    var duration: String? = null,

    @SerializedName("mbid")
    @Expose
    var mbid: String? = null,

    @SerializedName("url")
    @Expose
    var url: String? = null,

    @SerializedName("streamable")
    @Expose
    var streamable: Streamable? = null,

    @SerializedName("artist")
    @Expose
    var artist: Artist? = null,

    @SerializedName("image")
    @Expose
    var image: List<Image>? = null,

    @SerializedName("@attr")
    @Expose
    var attr: Attr? = null
)